### PR TITLE
TINKERPOP3-899: Bump to the latest version of Neo4j.

### PR DESCRIPTION
**Pull request by okram**
https://issues.apache.org/jira/browse/TINKERPOP3-899

This has been implemented and tested with `-DincludeNeo4j`. Note the `<exclusions>` I had to do in `<scope>test</scope>` as Neo4j doesn't use enforcer-plugin and they have self conflicts. I picked the highest version number of the self-conflict as the version to use.

Again, tested with `mvn clean install -DincludeNeo4j`.

My VOTE is +1.

### CODE REVIEW

File: neo4j-gremlin/src/main/java/org/apache/tinkerpop/gremlin/neo4j/structure/Neo4jHelper.java [hunk 1]
```diff
@@ -45,13 +44,16 @@ public static boolean isDeleted(final Neo4jNode node) {
...
+                return true;
+            else
+                throw e;
         }
     }
 
-    public static boolean isNotFound(RuntimeException ex) {
-        return ex.getClass().getSimpleName().equals("NotFoundException");
+    public static boolean isNotFound(final RuntimeException ex) {
+        return ex.getClass().getSimpleName().equals(NOT_FOUND_EXCEPTION);
```
Comment (dkuppitz) [inline neo4j-gremlin/src/main/java/org/apache/tinkerpop/gremlin/neo4j/structure/Neo4jHelper.java]: There's probably a good reason for the `toString()` comparison, but wouldn't it be easier/cleaner to do this:

```java
try {
    node.getKeys();
    return false;
} catch (NotFoundException e) {
    return true;
}
```

Comment (okram) [inline neo4j-gremlin/src/main/java/org/apache/tinkerpop/gremlin/neo4j/structure/Neo4jHelper.java]: `NotFoundException` is in Neo4j and that is not an Apache2 class :|.

**Comment by dkuppitz** [comment]
I see. Ugly, but functional.

+1

**Comment by PommeVerte** [comment]
Tested (int. tests too). Everything seems to be working.

+1

**Pull request merged**